src/resolve/resolved-dns-answer.c

```diff
@@ -656,6 +656,7 @@ void dns_answer_order_by_scope(DnsAnswer *a, bool prefer_link_local) {
 
 int dns_answer_reserve(DnsAnswer **a, size_t n_free) {
         DnsAnswer *n;
+        int r;
```

Comment:
nit: move declaration inside the ```if (*a)``` block

Reply:
"int r" is special, since it is the variable we store the error code of function calls in in 99% of the cases. it hence gets this special exception to always be placed in the outermost scope, at the top of the function, since it's not just any variable, but one that is an artifact how we do error handling everywhere after all... (if this makes any sense)

hence, I'd leave it like in @yuwata's existing version

Reply:
I see, that makes sense